.github/workflows/check.yml

```diff
@@ -1,12 +1,12 @@
 ---
-name: lint
+name: lint, and build
 
 on:
   push:
     branches:
       - main
-    tags:
```

Comment:
don't remove this, I left this blank on purpose so it gets triggered on every branch we may want to create.

Reply:
thinking this through a bit more, I'm unsure the whole `push` makes any sense given how we handle the event (we only lint the bash code & ensure the binary can be built); those actions - IMO -  fit the `pull_request` event,

I wouldn't even define the `push` event until we add some sort of publishing handler (e.g. publish container images for instance).

Reply:
So, here's my rationale behind this. At some point we may want to be a BDFL and create a push directly. I'd like for that to be linted. Same for tags, not applying now but I was thinking about 'cutting a release' if you know what I mean

Reply:
so, let's add that once it becomes a necessity. Right now, imo, there's just no reason for it.

Reply:
once we're done with this let's cut a release and add it